cmd/dep/status.go

```diff
@@ -652,13 +792,27 @@ func (bs *BasicStatus) marshalJSON() *rawStatus {
 	}
 }
 
+func (ds *DetailStatus) marshalJSON() *rawDetailStatus {
+	rawStatus := ds.BasicStatus.marshalJSON()
+
+	return &rawDetailStatus{
+		ProjectRoot:  rawStatus.ProjectRoot,
+		Constraint:   rawStatus.Constraint,
+		Locked:       formatDetailVersion(ds.Version, ds.Revision),
+		Latest:       formatDetailLatestVersion(ds.Latest, ds.hasError),
+		Source:       ds.Source,
+		Packages:     ds.Packages,
+		PackageCount: ds.PackageCount,
+	}
+}
+
 // MissingStatus contains information about all the missing packages in a project.
 type MissingStatus struct {
 	ProjectRoot     string
 	MissingPackages []string
 }
 
-func runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (hasMissingPkgs bool, errCount int, err error) {
+func (cmd *statusCommand) runStatusAll(ctx *dep.Ctx, out outputter, p *dep.Project, sm gps.SourceManager) (hasMissingPkgs bool, errCount int, err error) {
```

Comment:
Hi @zjs,

I think we should try to avoid refactoring things to our preference if possible. There's no real reason that this function needs to be assigned to `*statusCmd`, is there? I'm not a fan of the changes to the channel work below either, because I wonder if it doesn't make sense to just include the detail status as a field of basic status and again, reduce changes to what is truly necessary. Then again, we can always change it later if there is pushback from the project maintainers.

Reply:
Hi @zjs,

Regarding the changes to the channel workflow. I think I'm leaning towards what I said above re: pushback. Let's keep your design, and if the dep folks don't like it, we can revert it to the previous incarnation and change the composite object to a field in `BasicStatus`.

Reply:
> I think we should try to avoid refactoring things to our preference if possible. There's no real reason that this function needs to be assigned to *statusCmd, is there?

`runStatusAll` needs to be able to check `cmd.detail` to know whether to output "basic" or "detailed" information.

That's checked here:

https://github.com/akutz/dep/blob/5f52a57c47b1a9e5714736c0d266bd130154590c/cmd/dep/status.go#L1044

Is there a different way you'd recommend conveying that?

> Regarding the changes to the channel workflow. I think I'm leaning towards what I said above re: pushback. Let's keep your design, and if the dep folks don't like it, we can revert it to the previous incarnation and change the composite object to a field in BasicStatus.

Sounds good. I explored switching things around, but having `BasicStatus` contain the detailed information instead of the other way around makes some other things awkward. We can make if work, but let's see what they say.